src/main/ruby/truffleruby/core/truffle/string_operations.rb

```diff
@@ -411,5 +411,18 @@ def self.assign_regexp(string, index, count, replacement)
 
       Primitive.string_splice(string, replacement, bi, bs, enc)
     end
+
+    def self.on_codepoint_boundary?(string, byte_pos)
+      char_pos = Primitive.byte_index_to_character_index(string, byte_pos)
+      adjusted_byte_pos = if char_pos >= string.size
+                            # Handle index overflow cases.
+                            # @see com.oracle.truffle.api.strings.ByteIndexToCodePointIndexNode#execute for details.
```

Comment:
@andrykonchin You had a question here - I rather added the explanation to the source so it's more discoverable: https://github.com/oracle/graal/pull/6635/files